### PR TITLE
fix: cache metadata by alias

### PR DESCRIPTION
**Pull request by PowderLi**
issue: #34161

1. database --> collectionName --> collectionID && database --> collectionID --> collectionInfo
2. same as shardLeaders
3. expire different metadata in different situations

### CODE REVIEW

File: internal/proxy/meta_cache.go [hunk 1]
```diff
@@ -251,19 +249,20 @@ type MetaCache struct {
...
-	userToRoles      map[string]map[string]struct{}          // user to role cache
-	mu               sync.RWMutex
-	credMut          sync.RWMutex
-	leaderMut        sync.RWMutex
-	shardMgr         shardClientMgr
-	sfGlobal         conc.Singleflight[*collectionInfo]
-	sfDB             conc.Singleflight[*databaseInfo]
+	dbInfo         map[string]*databaseInfo                         // database -> db_info
+	collNameInfo   map[string]map[string]typeutil.UniqueID          // database -> collectionName -> collection_id
+	collInfo       map[string]map[typeutil.UniqueID]*collectionInfo // database -> collectionName -> collection_info
```
Comment (longjiquan) [inline internal/proxy/meta_cache.go:254]: database -> collectionID -> collection_info

File: internal/proxy/meta_cache.go [hunk 2]
```diff
@@ -251,19 +249,20 @@ type MetaCache struct {
...
+	collShard      map[string]map[string]typeutil.UniqueID          // database -> collectionName -> collection_id
+	collNameShard  map[string]map[typeutil.UniqueID]*shardLeaders   // database -> collectionName -> collection_leaders
```
Comment (longjiquan) [inline internal/proxy/meta_cache.go:256]: Ditto.

File: internal/rootcoord/alter_collection_task.go [hunk 3]
```diff
@@ -58,23 +59,16 @@ func (a *alterCollectionTask) Execute(ctx context.Context) error {
...
-		req:      a.Req,
-		core:     a.core,
-	})
+	if err := a.core.meta.AlterCollection(ctx, oldColl, newColl, a.GetTs()); err != nil {
+		return err
+	}
 
-	return redoTask.Execute(ctx)
+	a.Req.CollectionID = oldColl.CollectionID
+	return a.core.broker.BroadcastAlteredCollection(ctx, a.Req)
```
Comment (longjiquan) [inline internal/rootcoord/alter_collection_task.go:71]: I think we should redo `BroadcastAlteredCollection`.

Comment (PowderLi) [reply internal/rootcoord/alter_collection_task.go:71]: there is no difference between use `redoTask.AddSyncStep(&BroadcastAlteredCollectionStep`

Comment (longjiquan) [reply internal/rootcoord/alter_collection_task.go:71]: I mean it shouldn't be a sync step.

**Comment by Peter-YoungUk** [comment]
Is there any reason to close this PR? 
I believe it could cause performance issues when using aliases with search operations, especially under high traffic. 
This is due to cache misses, which lead to every round trip from the proxy to the coordinator.

**Pull request closed without merging**